### PR TITLE
feat: Bump ic-js version and support new topics, while not showing them when editing following

### PR DESCRIPTION
**Pull request by jasonz-dfinity**
# Motivation

2 new topics are added to NNS Governance while following on them are not allowed yet. Before NNS Governance can start allowing following to be set on them, we need to make sure NNS Dapp can render them correctly. Because of the same reason, we do not show the new topics when editing the followees.

# Changes

* Bump ic-js version
* Define en versions of the topic titles and descriptions
* Exclude the new topics when editing following

# Tests

* unit tests 
* manual tests in the local environment
  * followees on the new topics can be rendered
  * new topics do not show up on the edit followees panel
Screenshot 2024-07-18 at 1 32 24 PM
Screenshot 2024-07-18 at 1 32 43 PM

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

File: frontend/src/lib/constants/neurons.constants.ts [hunk 1]
```diff
@@ -20,6 +20,14 @@ const FIRST_TOPICS = [
   Topic.SnsAndCommunityFund,
 ];
 const LAST_TOPICS = [Topic.ExchangeRate];
+
+// Topics that neurons cannot yet set following for.
+// TODO: Remove this list when the NNS Governance supports following on those topics.
```
Comment (dskloetd) [inline frontend/src/lib/constants/neurons.constants.ts:25]: Would you expect that with future topics we would do the same thing?
In that case, would it make sense not to remove this but just keep it as empty later?

File: frontend/src/lib/i18n/en.json [hunk 2]
```diff
@@ -366,6 +366,10 @@
     "topic_15_subtitle": "Proposals related to the management of API boundary nodes",
     "topic_16_title": "Subnet Rental",
     "topic_16_subtitle": "All proposals related to renting a subnet, for example a subnet rental request.",
+    "topic_17_title": "Protocol Canister Management",
+    "topic_17_subtitle": "Proposals related to managing canisters that are considered as part of the IC protocol.",
+    "topic_18_title": "Service Nervous System Management",
+    "topic_18_subtitle": "Proposals related to managing the service nervous system (SNS).",
```
Comment (dskloetd) [inline frontend/src/lib/i18n/en.json:372]: "the" in "the service nervous system" suggests there is only 1.
Is this a typo?

File: frontend/src/lib/i18n/en.json [hunk 3]
```diff
@@ -366,6 +366,10 @@
     "topic_15_subtitle": "Proposals related to the management of API boundary nodes",
     "topic_16_title": "Subnet Rental",
     "topic_16_subtitle": "All proposals related to renting a subnet, for example a subnet rental request.",
+    "topic_17_title": "Protocol Canister Management",
+    "topic_17_subtitle": "Proposals related to managing canisters that are considered as part of the IC protocol.",
+    "topic_18_title": "Service Nervous System Management",
```
Comment (dskloetd) [inline frontend/src/lib/i18n/en.json:371]: This seems confusing.
We already have topic 14: "SNS & Neurons' Fund"
Are they really different? If so, can we make it clearer why they are not the same?

File: frontend/src/tests/lib/utils/neuron.utils.spec.ts [hunk 4]
```diff
@@ -2166,30 +2167,38 @@ describe("neuron-utils", () => {
...
 
-    it("should not return deprecated topics", () => {
+    it("should not return deprecated or disabled topics", () => {
       expect(topicsToFollow(neuronWithoutManageNeuron)).toEqual(
         TOPICS_TO_FOLLOW_NNS.filter(
           (topic) =>
-            topic !== Topic.ManageNeuron && !DEPRECATED_TOPICS.includes(topic)
+            topic !== Topic.ManageNeuron &&
+            !DEPRECATED_TOPICS.includes(topic) &&
+            !TOPICS_WITH_FOLLOWING_DISABLED.includes(topic)
```
Comment (dskloetd) [inline frontend/src/tests/lib/utils/neuron.utils.spec.ts:2176]: This is (and was) just a reimplementation of the logic. That doesn't seem very useful as a test. Can we hard-code the explicit list we expect here?
(and below)

**Review by dskloetd — COMMENTED** [review]
Have you tested manually what is and isn't possible and if things are rendered correctly?

File: frontend/src/tests/lib/utils/neuron.utils.spec.ts (hunk 4 above)
Comment (jasonz-dfinity) [reply frontend/src/tests/lib/utils/neuron.utils.spec.ts]: Done.

File: frontend/src/lib/i18n/en.governance.json [hunk 5]
```diff
@@ -212,4 +212,4 @@
     "DeployHostosToSomeNodes": "Deploy a HostOS version to a given set of nodes. The proposal changes the HostOS version that is used on the specified nodes.",
     "SubnetRentalRequest": "A proposal to rent a subnet on the Internet Computer.<br/><br/>The Subnet Rental Canister is called when this proposal is executed, and the rental request is stored there. The user specified in the proposal needs to make a sufficient upfront payment in ICP in order for the proposal to be valid, and the subnet must be available for rent. The available rental conditions can be checked by calling the Subnet Rental Canister."
   }
-}
+}
```
Comment (dskloetd) [inline frontend/src/lib/i18n/en.governance.json:215]: Please don't remove the trailing line break.
You should probably fix the config of your editor to not do this.

File: frontend/src/lib/i18n/en.json [hunk 6]
```diff
@@ -1032,4 +1032,4 @@
     "show_all": "Show all",
     "import_token": "Import Token"
   }
-}
+}
```
Comment (dskloetd) [inline frontend/src/lib/i18n/en.json:1035]: again

File: CHANGELOG-Nns-Dapp-unreleased.md
Comment (dskloetd) [inline CHANGELOG-Nns-Dapp-unreleased.md:1]: Go ahead and merge after testing manually.

**Review by dskloetd — APPROVED** [review]

File: CHANGELOG-Nns-Dapp-unreleased.md
Comment (jasonz-dfinity) [reply CHANGELOG-Nns-Dapp-unreleased.md:1]: Tested manually and updated the PR description with screenshots

File: frontend/src/lib/i18n/en.json (hunk 6 above)
Comment (jasonz-dfinity) [reply frontend/src/lib/i18n/en.json]: Done

File: frontend/src/lib/i18n/en.governance.json (hunk 5 above)
Comment (jasonz-dfinity) [reply frontend/src/lib/i18n/en.governance.json]: Done.

**Pull request merged**